server/src/application/application.service.ts

```diff
@@ -11,6 +12,20 @@ export class ApplicationService {
     private applicationRepository: Repository<Application>
   ) {}
 
+  async getApplications(query: GetApplicationsDto): Promise<[Application[], number]> | null {
+    const queryBuilder = this.applicationRepository.createQueryBuilder('app');
+
+    if (query.facilityName)
+      queryBuilder.andWhere('app.facilityName = :facilityName', {
+        facilityName: query.facilityName,
+      });
+    if (query.orderBy) queryBuilder.orderBy(`app.${query.orderBy}`, query.order);
+    query.filter(queryBuilder);
+
+    const applications = await queryBuilder.getManyAndCount();
+    return [applications[0], applications[1]];
```

Comment:
Use a common RO for paginated things. Refer the codebase already shared!

Reply:
Used new PaginationRo as the return instead.